### ISSUE TITLE
Adding background hover css class displays default theme

### ISSUE THREAD

**Issue by PowerHauer** [OfficeDev/office-ui-fabric-react#2351]
### Bug Report

- __Package version(s)__:
office-ui-fabric-react@2.31.0
- __Browser and OS versions__:
All browsers
(Tested on Edge, Chrome 59 and Firefox 54)

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? No

Requested priority: Normal 

Products/sites affected:
SPFx

#### Describe the issue:

I am developing a SPFx WebPart with theming in mind.
I added an icon and gave it fabric ui classes for theming:

`<i className="ms-Icon ms-Icon--ChevronRightSmall ms-fontColor-themePrimary ms-fontColor-white--hover ms-bgColor-themePrimary--hover"
                    aria-hidden="true"></i>`

The icon is displayed in the theme color. When the mouse hovers over the element, the icon is white and the background is colored blue (from the default theme).

#### Actual behavior:

The icon is displayed in the theme color. When the mouse hovers over the element, the icon is white and the background is colored blue (from the default theme).

![iconswithouthover](https://user-images.githubusercontent.com/30617969/28814549-56d3b1e4-769e-11e7-9b2d-c67a5e96e556.PNG)

![iconswithhover](https://user-images.githubusercontent.com/30617969/28814568-68f4b80a-769e-11e7-8f6c-dcba629803f1.PNG)

#### Expected behavior:

The icon is displayed in the theme color. When the mouse hovers over the element, the icon is white and the background should be pink (from the current theme).

**Comment by micahgodbolt** [comment]
@mikewheaton this is a core issue, correct?

**Comment by mikewheaton** [comment]
While the [colors Fabric Core](http://dev.office.com/fabric#/styles/colors) are named `theme`, the CSS classes don't actually tie into the theming system in Fabric React. When it comes to Fabric Core you can consider `themePrimary` to be `bluePrimary`. There's work underway for the components to be styled with [Glamor](https://github.com/threepointone/glamor), which would allow you to pass in an object to set the `<Icon>` component's text color to the primary theme color. I'm not sure when this will be available.

**Comment by PowerHauer** [comment]
I just tried to go along the descriptions on the website to Fabric UI.
https://dev.office.com/fabric#/styles/colors

There it is described to achieve a different color (foreground, background or text) while hovering to use the classes with the suffix "--hover".
My assumption was that this works with the native office 365 theming settings, since all non "--hover" classes work like in the example above. If the theme is pink, everything with "themePrimary" is pink and so on. This works even for all other theme dependent classes: "themePrimary", "themeDarker", "themeDark", ... Except for elements with the "--hover" classes, such as "ms-bgColor-themeXXX--hover".

So basically theming is supported, except for "--hover" classes, which use the default blue theme?

**Comment by mikewheaton** [comment]
Theming is supported in Fabric React through the Styling package (see [documentation](http://dev.office.com/fabric#/components/themes)) but is not supported by the [Fabric Core classes](http://dev.office.com/fabric#/styles/colors).

**Comment by erichdev** [comment]
Closing. Please reopen if you have additional questions
